.github/workflows/build.yml

```diff
@@ -213,17 +213,36 @@ jobs:
           echo Downloading Terraria Server zip
           curl -s -L https://terraria.org/api/download/pc-dedicated-server/terraria-server-$TERRARIA_VERSION.zip -o terrariaserver.zip
           unzip -q terrariaserver.zip
-          echo Extracting Setup Secret Assets
-          gpg --quiet --batch --yes --decrypt --passphrase="$SETUP_ASSETS_PASSPHRASE" --output ./$TERRARIA_VERSION/Windows/Terraria.exe ./setup/SecretAssets/SetupAsset1.gpg
-          gpg --quiet --batch --yes --decrypt --passphrase="$SETUP_ASSETS_PASSPHRASE" --output ./$TERRARIA_VERSION/Windows/SteelSeriesEngineWrapper.dll ./setup/SecretAssets/SetupAsset2.gpg
           
           echo Installing XNA
           curl -s -L https://github.com/JavidPack/tModLoaderPublishIntegration/raw/master/xnafx40_redist.msi -O
           msiexec //i xnafx40_redist.msi //quiet
           echo XNA Installed
           
-          echo ::group::Running setup.bat
-          ./setup.bat auto ./$TERRARIA_VERSION/Windows
+          echo ::group::Running setup-cli.bat
+          if [ ${{github.ref}} = 'refs/heads/${{env.StableBranch}}' ] ; then
+            tmlVersionDefine=TML_$(cut -d '.' -f1 <<< $TMLVERSION)_$(cut -d '.' -f2 <<< $TMLVERSION)
+            echo "TMLVERSIONDefine: $tmlVersionDefine"
+            sed -i "s#<\!-- TML stable version define placeholder -->#<DefineConstants>\$(DefineConstants);$tmlVersionDefine</DefineConstants>#g" ./patches/tModLoader/Terraria/release_extras/tMLMod.targets
+          fi
+
+          if [ ! -d "./src/decompiled" ]; then
+            ./setup-cli.bat decompile --key $TERRARIA_OWNERSHIP_KEY --terraria-steam-dir ./$TERRARIA_VERSION/Windows --tml-dev-steam-dir steam_build --plain-progress --max-parallelism 1 -f --strict
+
+            err=$?
+            if [ $err -ne 0 ]; then
+              exit $err
+            fi
```

Comment:
Github actions runs the shell with `-e` so there's no need to handle exit codes yourself.